app/utils/auth.ts

```diff
@@ -10,23 +12,58 @@ interface CustomUser extends User {
   };
 }
 
+/**
+ * Get the allowed roles array from a user object
+ * */
+export const getRolesArray = (user: CustomUser | undefined) =>
+  user?.["https://hasura.io/jwt/claims"]?.["x-hasura-allowed-roles"];
+
 /**
  * Get a user's hasura role name from their allowed roles
  */
-export const getHasuraRoleName = (user: CustomUser | undefined): string => {
-  const roles =
-    user?.["https://hasura.io/jwt/claims"]?.["x-hasura-allowed-roles"];
-
+export const getHasuraRoleName = (roles?: string[]): string => {
   if (!roles) {
     return "";
   } else if (roles.includes("vz-admin")) {
     // todo
     // If user has more than one role, it is because they have `itSupervisor` and `vz-admin`
-    // `vz-admin` is the Hasura rolename, `itSupervisor` is some legacy name from back when
-    // we used the hasura `admin` role instead of a named role for admins. John is pretty sure
-    // sure we stop using `itSupervisor`
+    // `vz-admin` is the Hasura rolename, `itSupervisor` is a legacy role that was previously
+    // used to enable a user to create and edit other admins. this app will do away with
+    // that behavior: any admin can create and modify other admins. A follow-up task is
+    // to clean up the Auth0 user database that references `itSupervisor`
```

Comment:
IIRC it was originally a request from Lewis to be able to have a super-admin role, but I don't think it's worth the technical overhead to maintain this. I think we can safely get by with three tiers:

* Read only
* Editor: has all editing access except create/edit users
* Admin: has all editing access and can create/edit users

Reply:
Full agreement. We have access to the whole stack, top to bottom, so there's no reason that I can conceive of that we need to build special capabilities into the front-end UI that we can't do with our backend access.